TempMail/utils/Email.swift

```diff
@@ -10,31 +10,27 @@
 import SwiftUI
 
 class Email : ObservableObject{
-    @Published var email_addr: String
-    @Published var inbox: [InboxModel]
-    var requestUrl: URLComponents
+    @Published var email_addr: String?
+    @Published var inbox: [InboxModel]?
     
     //  init method
     init() {
-        self.email_addr = ""
-        self.inbox = [InboxModel]()
-        requestUrl = URLComponents()
-        requestUrl.scheme = "https"
-        requestUrl.host = "1secmail.com"
-        requestUrl.path = "/api/v1/"
+        reset()
+    }
+    
+    //resets the email
+    func reset() -> Void {
+        self.inbox = nil
+        self.email_addr = setNewEmailAddr()
         loadInbox()
-        setNewEmailAddr()
     }
     
     //  generates an email address
-    func setNewEmailAddr() -> Void {
+    func setNewEmailAddr() -> String {
```

Comment:
change to getNewEmailAddr()

Reply:
Why? I believe the purpose of this function is to set a new email address, not to fetch it. getEmailAddr() and getNewEmailAddr might be too similar in name as well, which is another issue in my opinion. If you don't believe this is a set method, I could change the func signature to generateEmailAddr, or generateNewEmailAddr. What do you think?

Reply:
Good point, genEmailAddr is a better name

Reply:
Actually, genNewEmailAddr is better, its more explicit